### PR TITLE
Create java-compatibility.adoc [TOMEE-2540]

### PR DESCRIPTION
**Pull request by dineshkumarg**
This is a live document until TomEE 8.0.0 is released. Till that point of time, any updates (either additions/removal) of Java versions support can be done to this document

### CODE REVIEW

File: docs/java-compatibility.adoc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+= Java Compatibility with TomEE 8.0.x
+:index-group: General Information
+:jbake-type: page
+:jbake-status: published
+
+TomEE 8.0.x is Java EE 7, Java 8 & Java 10. Any OS that supports Java 10 works
```
Comment (Daniel-Dos) [inline docs/java-compatibility.adoc]: change Java EE 7 to Java EE 8

Comment (gilbertoca) [reply docs/java-compatibility.adoc]: Java 10 ended support in September 2018, there is no more release for it.

Comment (dineshkumarg) [reply docs/java-compatibility.adoc]: Updated with correct version 
Thanks

**Comment by dineshkumarg** [comment]
@cesarhernandezgt when you have time, can you please have a review on this and let me know if any changes are required, else please approve
Thanks

**Comment by cesarhernandezgt** [comment]
btw, @dineshkumarg it's useful to have the JIRA ticket ID as part of the GitHub issue title to keep nice integration and easiness of navigation flow when we are reviewing PR :).

**Comment by dineshkumarg** [comment]
> btw, @dineshkumarg it's useful to have the JIRA ticket ID as part of the GitHub issue title to keep nice integration and easiness of navigation flow when we are reviewing PR :).

taken care @cesarhernandezgt

**Comment by dineshkumarg** [comment]
> My only suggestion would be to change 8.0.x and 8.0.0 with 8.0.0-M3 . Since we don't have a TomEE 8.0.0 release yet and this documentation will end up been published on the website.
> 
> Also a note like the one we have on the /download page looks good:
> 
> `Note: Please note the 8.0.0-M3 release is a milestone release intended for evaluation purposes and should not be used in production.`

@cesarhernandezgt can you post me full URL of /download page, I couldn't edit this one https://tomee.apache.org/download-ng.html

File: docs/java-compatibility.adoc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+= Java Compatibility with TomEE 8.0.0-M3
+:index-group: General Information
+:jbake-type: page
+:jbake-status: published
+
+TomEE 8.0.0-M3 is Java EE 8 & Java 11. Any OS that supports Java 11 works
```
Comment (cesarhernandezgt) [inline docs/java-compatibility.adoc]: As far as I know, TomEE 8.0.0-M3 is still not built using Java 11.

**Comment by cesarhernandezgt** [comment]
@dineshkumarg 
It must be a misspelling from my side. My above comment about the /download page is not related to updating the actual /download page, it's intended for adding the quoted text as part of the text your PR is introducing.

File: docs/java-compatibility.adoc (hunk 2 above)
Comment (dineshkumarg) [reply docs/java-compatibility.adoc]: So, TomEE 8.0.0-M3 is only Java EE 8 as of now?

Comment (rzo1) [reply docs/java-compatibility.adoc]: But it works fine @ runtime with Java 11even if it is not build with it :)

Comment (cesarhernandezgt) [reply docs/java-compatibility.adoc]: Hi @rzo1 , Base on a couple of thread on the mailing list [1]. Current TomEE M3 and therefore master branch doesn't fully compile nor pass yet all the test when runs on Java 11. The impression TomEE M3 runs on Java 11 may be true for a couple of apps but the server is not yet fully Java 11 compatible. That's why I think advertising on the website that TomEE M3 is Java 11 compatible may be misleading for the reader.

[1] https://lists.apache.org/list.html?dev@tomee.apache.org:lte=6M:java%2011

Comment (dineshkumarg) [reply docs/java-compatibility.adoc]: @cesarhernandezgt please check the document where I made changes, if the statements under Note are good. Or suggest a better notes which can be published

Thanks

Comment (cesarhernandezgt) [reply docs/java-compatibility.adoc]: done, I marked as +1, thanks much :)

**Comment by dblevins** [comment]
Looks great.  Really fantastic to see this kind of thing get documented and nailed down.  Small but important step.

Thank you @dineshkumarg !

**Comment by dineshkumarg** [comment]
Thanks @dblevins @cesarhernandezgt @rzo1 @Daniel-Dos @gilbertoca for your help and support

**Pull request merged**